app/services/pieces_justificatives_service.rb

```diff
@@ -37,4 +33,21 @@ def self.upload_one! dossier, user, params
 
     piece_justificative
   end
+
+  def self.save_pj(content, dossier, tpj, user)
+    pj = PieceJustificative.new(content: content,
+                                dossier: dossier,
+                                type_de_piece_justificative: tpj,
+                                user: user)
+
+    pj.save ? '' : "le fichier #{pj.libelle} n'a pas pu être sauvegardé<br>"
+  end
+
+  def self.missing_pj_error_messages(dossier)
+    mandatories = dossier.types_de_piece_justificative.select(&:mandatory)
+    presents = dossier.pieces_justificatives.map(&:type_de_piece_justificative)
+    missing = mandatories.reject { |m| m.in?(presents) }
+
+    missing.map { |m| "Le document #{m.libelle} doit être fourni.<br>" }
```

Comment:
Mettre les adjectif au singulier et appeler les variables `x_pjs` ? et itérer sur `pj` et pas `m` ?